### ISSUE TITLE
[BUG] Re-scan Files with Scene Numbering enabled 

### ISSUE THREAD

**Issue by heymoe** [SickRage/sickrage-issues#673]
Branch: Master
Commit hash: 3f0efd1338faf8796fd355bada4f546ec1cb5ed6
Your operating system and python version: CentOS 5.11 / Python 2.7

American Dad! is one of those shows where the episode numbering on theTVDB and what the scene uses are all out of sync.

Example: theTVDB: Season 8 Episode 1 = Scene: Season 7 Episode 1

XEM Link for episode mappings: http://thexem.de/xem/show/2913

So using the example above, s08e01 (tvdb) is wanted. SR set to Scene Numbering looks up in XEM and finds that s08e01 (tvdb) = s07e01 (scene) so it searches and finds that available. SR triggers the download of s07e01 (scene), processes the download and places the new file in the "Season 08" folder as "American Dad! - s08e01.mp4" (tvdb). Everything is working as expected so far. This is the expected behavior. 

The Problem: So, SR has now successfully found and downloaded the correct file (s07e01 - scene) for s08e01 (tvdb) that was wanted and placed it in the correct directory and named it properly on the filesystem. 

If you now click "Re-Scan Files", SR sees the file "American Dad! - s08e01.mp4" on the file system in the "Season 08" folder and starts processing that file again as part of the "Re-Scan". Refreshing the "American Dad!" page in SR once the Re-Scan is done, SR will now show that s09e01 has been found and if you hover over the episode # to see what the path is to the file for the episode (s09e01) it shows "Season 08/American Dad! - s08e01.mp4". s08e01 (scene) = s09e01 (tvdb) 

Basically SR seems to be applying the XEM lookup on the files on the file system as if those files were scene episodes and not already processed and sorted tvdb episodes which is causing episodes that haven't actually been downloaded to be marked as downloaded.

What I expected is that SR would see the already downloaded and processed file on the file system as being the episode it says it is (tvdb) and process it as such instead of thinking it found a new file that it hasn't seen before and process it like it was a scene numbered episode which would translate that file to the wrong episode. 

I hope this all makes sense. If not let me know and I'll try and explain the problem better.. Thanks!

**Comment by SickRage2** [comment]
Thanks for the issue report! Before a real human comes by, please make sure your report has all the below criteria checked

- [ ] Include basic information: Branch/Commit, OS, What you did, What happened, What you expected
- [ ] Enable debug logging (be sure to disable after the bug is fixed)
- [ ] Post debug logs, either inline (for smaller logs) or using gist)

Please make sure you also read [how to create an issue](https://github.com/SickRage/sickrage-issues/blob/master/README.md) and followed all of the steps. 

The title should describe your issue. Having "SR not working" or "I get this bug" for 100 issues, isn't really helpful. We will close issues if there isn't enough information.

Sometimes the devs may seem like grunts and respond with short answers. This isn't (always) because the dev hates you, but because he's on mobile or busy fixing bugs. If something isn't clear, please let us know, and this bot may get updated to automatically answer you.

Thanks!

**Comment by neoatomic** [comment]
Sadly the tvdb/scene conversions aren't perfect, but this is about the best it can do.
When sickrage downloads an episode using the conversion it stored that data in the database.
However if you remove the episode or even the database that data gets lost. 
And so there are some more difficulties like re-scanning if you renamed files etc.

If you have some solid suggestions/improvements then feel free to open a feature request.: 
http://feathub.com/SickRage/SickRage
Hopefully we will be able to implement that in the future.

**Comment by heymoe** [comment]
Well the problem is that I didn't do any renaming or manual manipulation of the episode.

SR download the correct scene (s07e01) -> tvdb (s08e01) episode
I hit Re-Scan files 
SR now says I have another episode that I don't really have (s09e01) because it saw the s08e01 file it just downloaded and processed and applied the scene -> tvdb conversion on it to get s09e01.

Sure I can see if I manually copied the s08e01 file into the SR dir and it never saw it before that it could cause problems but this is a file it already processed, had in it's DB and it decided to link it to the s09e01 episode while keeping it listed as s08e01 too. 

From what you're saying if SR was the only thing to every manage a file it should not do what I'm seeing it do right now. 

This bug or what not does make it convoluted to mass import a bunch of episodes for a series like this. I'd have to rename them all to the scene numbering and then tell SR to manually post-process them so it will then convert them from scene to tvdb numbering to get them into SR correctly and I'll happily do this to get them imported correctly but if I ever click the Re-Scan Files on this show it will screw up everything the way it is right now.

I found this issue because I noticed I wasn't getting any of the new Season 13 episodes and that was because SR saw I had Season 12 and linked those to Season 13 so it thinks I have them already.

**Comment by neoatomic** [comment]
I see what you are saying. 
But in a normal situation you would not use the rescan button as SR knows whats downloaded and in what quality. Its more meant as a way to solve issues, and not for daily usage.
And we do not advice to import any episode files directly in the shows folder. Put them in the PP folder and let Sickrage process them as if they where "downloaded". That would also prevent any of the issues you currently have. As basically you are not using Sickrage correctly by importing/re-scanning them manually.

The only other way that might work is disable scene numbering when you hit the Rescan button. That should give you the tvdb episode scheme. but you will have tot est that.

**Comment by heymoe** [comment]
I typically don't use the Re-Scan Files features except to solve problems like you said and I was trying to solve a problem with American Dad which by Re-Scanning Files made everything worse.

I totally get what you're saying about how SR should be used and 99% of the time that's how I use it. 

This wasn't a problem or didn't seem to be a problem a few months ago when tvdb decide to flip the episode numbering of Mythbusters upside down but using year seasons. Mythbusters in my SR were all added up to that post using the original, non year seasons. After tvdb make there change and SR refresh it's tvdb data, SR was all sorts of messed up. I manually renamed all my files to the new year seasons, put them into the new year season sub-folders and had SR Re-Scan Files. Once it was done, I everything was happy again. Mythbusters is another one of those shows where scene and tvdb numbering have been messed up for a while before this year season change came about so I already had it setup with scene number checked. So I'm not sure if something changed with the Re-Scan File code or not between then and now.

Any ways, disabling scene numbering and re-scan files seems to do the trick and turning scene number back on after that seems to be fine too. I guess I'll have to wait and see what happens when the next new episode comes out to make sure things are really working.

Honestly, to me, the recommended fix is to make the "Re-Scan Files" scan the show's SR directory with scene number turned off no matter if it's turned on or not. It doesn't make sense for someone to be manually manipulating files in the show's SR directory in a numbering scheme that is not what SR is using in that directory. Why would I copy a file into that directory that is already full of tvdb numbered episodes with an episode that is scene numbered and then expect SR to know that that one file is scene numbered and it should fix it for me. "Re-Scan Files" is not Post-Processing. If you want to import a file that is scene numbered, us the manual post-processing function. 

For now I guess if I ever need to us Re-Scan Files on any show that has scene numbering enabled, I'll just disable it first as an important safety tip.

**Comment by 28064212** [comment]
I'm pretty sure this is incorrectly closed. It has worked properly on previous versions of SR. SR now appears to be using the scene exceptions to match to the post-processed files, when it should be using the tvdb info.

SR uses scene numbering to _search_ for files, but when it post-processes them, it uses indexer numbering to save them e.g. for American Dad, it searches for 12x1, saves it as 13x1. So when SR re-scans files, it should look for 13x1, it doesn't make sense that it's now looking for 12x1

**Comment by dexhain** [comment]
I'm pretty sure this is incorrectly closed. It has worked properly on previous versions of SR. SR now appears to be using the scene exceptions to match to the post-processed files, when it should be using the tvdb info.

SR uses scene numbering to search for files, but when it post-processes them, it uses indexer numbering to save them e.g. for American Dad, it searches for 12x1, saves it as 13x1. So when SR re-scans files, it should look for 13x1, it doesn't make sense that it's now looking for 12x1

**Comment by heymoe** [comment]
Agreed re: dexhain

**Comment by neoatomic** [comment]
@miigotu Any thoughts on this issue.?

**Comment by miigotu** [comment]
I'm pretty sure this is a bug, but it has been a bug for a REALLY long time. No ETA

**Comment by yongbi85** [comment]
Yeah i can confirm American Dad is THE show that provokes so many bugs with sickrage. I ended up having a copy of all american dad seasons on another hard drive than SR because i need to copy them over every 2 or 3 days as Sickrage keeps messing the serie up (trying to download stuff i already have in preferred quality, overwriting season 8 with season 7 , or deleting the season 7 episode 1 when download season 8 episode 1, ....)

**Comment by heymoe** [comment]
So it seems that even though I got SR to import / add all the American Dad shows that I had in tvdb numbering with scene numbering turned off without any problems. Turning it on after the import seem to be OK but it looks like the next time SR goes and does it's automated TV Show refresh, it decided to reprocess / re-scan the files already downloaded and stored in tvdb numbering as being scene numbering so now SR has Season 13 fully populated with Season 12 files as an example. Season 13 being the current episode where only the first episode has aired so far so now SR thinks it has them all so it won't grab future real Season 13 episodes. 

So basically this issue is not limited to the manual clicking on the "Re-Scan Files" button but any time SR does a show refresh which it typically does every 24 hours.

**Comment by miigotu** [comment]
We get what the problem is, no further info needed. It will get fixed if/when we figure it out.

**Comment by mgaulton** [comment]
I've commonly see this with shows like Kitchen Nightmares and Cutthroat Kitchen and the like. most others have been ok.

**Comment by Wob76** [comment]
Has there been any progress with this issue? I have resorted to excluding shows from sickrage as it keeps seeming old episodes as new ones and not downloading them.

**Comment by heymoe** [comment]
Not that I'm aware of. I've had American Dad! paused in SR since I've reported this bug and I've been manually doing the processing for it myself. 

I did find that even if I let SR do all the work and I never click on the "Re-Scan Files" button that eventually SR will trigger this bug on its own when it does a show update / maintenance itself. So, really there is no option to have SR auto process this show right now. 

American Dad! seems to be the main show that people see this bug in but I'm sure it would be a problem for just about any show that has Scene Numbering enabled. 

This bug has been open for about 4 months now so hopefully it's moving up on the list of things to fix soon.

**Comment by dexhain** [comment]
For American Dad, what I do is:
- edit the show to NOT use "Scene Numbering", and set default episode staus to "skipped"
- after an episode has been released, change it to Scene Numbering, then set the new episode to Wanted
- after it's been located and downloaded, set the show back to NOT using "Scene Numbering"

It's not ideal, but it works

**Comment by heymoe** [comment]
Yeah, that's one way of doing it. I just manually download the file, rename it, copy to the correct directory and click the "Re-Scan Files" button. SR has "Scene Numbering" turned off at this point for me. 

Different steps but the end results are the same.

**Comment by mgaulton** [comment]
I'd like to see fallback similar to season vs episode searching. "did i find the scene numbered file, then try the indexer numbered file" sort of thing.

**Comment by Wob76** [comment]
Has there been any progress on this bug?

**Comment by heymoe** [comment]
None that I'm aware of.

**Comment by heymoe** [comment]
Do I need to re-submit this issue into the current SickRage issues group?

**Comment by 28064212** [comment]
Yes, still an issue

**Comment by heymoe** [comment]
For the people following this issue. I did re-submit and it looks like there is a fix now so check it out.

https://github.com/SickRage/SickRage/issues/2996
https://github.com/SickRage/SickRage/pull/3004
